server/tests/test_server_pages.py

```diff
@@ -0,0 +1,25 @@
+import json
+import re
+from server.notebooks.models import Notebook
+
+from django.urls import reverse
+
+
+def _get_page_data(page_content_str):
+    m = re.search('window.pageData = (\{.*\});', str(page_content_str))
+    return json.loads(m.group(1))
+
+
+def test_index_view(client, two_test_notebooks):
+    resp = client.get(reverse('index'))
+    assert resp.status_code == 200
+    assert set(_get_page_data(str(resp.content)).keys()) == set(['userInfo',
+                                                                 'notebookList'])
+
+
+def test_user_view(client, fake_user, two_test_notebooks):
```

Comment:
Seems like a reasonable format to me, I see how we can use this to test when a username doesn't exist.

Reply:
Yep, here's an example of such a test in the current repo (note that the directory will change after this lands): 

https://github.com/iodide-project/iodide/blob/5ad54f81e97d4dbadaf645a47eb92615e30fb273/server/notebooks/tests/test_notebook_revision_api.py#L53